sdk/spring/azure-spring-cloud-stream-binder-servicebus-core/src/main/java/com/azure/spring/servicebus/stream/binder/ServiceBusMessageChannelBinder.java

```diff
@@ -59,9 +57,7 @@ protected MessageHandler createProducerMessageHandler(
         handler.setSendFailureChannel(errorChannel);
         if (producerProperties.isPartitioned()) {
             handler.setPartitionKeyExpressionString(
-                    "'partitionKey-' + headers['" + BinderHeaders.PARTITION_HEADER + "']");
-        } else {
-            handler.setPartitionKeyExpression(new FunctionExpression<Message<?>>(m -> m.getPayload().hashCode()));
```

Comment:
Remove the default value of partitionkey is a breaking change, we should clarity it in the changelog.